src/badge/examples/EthereumYearBadge.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {Attestation} from "@eas/contracts/IEAS.sol";
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+import {ScrollBadge} from "../ScrollBadge.sol";
+import {ScrollBadgeCustomPayload} from "../extensions/ScrollBadgeCustomPayload.sol";
+import {ScrollBadgeNoExpiry} from "../extensions/ScrollBadgeNoExpiry.sol";
+import {ScrollBadgeNonRevocable} from "../extensions/ScrollBadgeNonRevocable.sol";
+import {ScrollBadgeSelfAttest} from "../extensions/ScrollBadgeSelfAttest.sol";
+import {ScrollBadgeSingleton} from "../extensions/ScrollBadgeSingleton.sol";
+
+string constant ETHEREUM_YEAR_BADGE_SCHEMA = "uint256 year, bytes signature";
+
+function decodePayloadData(bytes memory data) pure returns (uint256, bytes memory) {
+    return abi.decode(data, (uint256, bytes));
+}
+
+/// @title EthereumYearBadge
+/// @notice A badge that represents the year of the user's first transaction on Ethereum.
+contract EthereumYearBadge is
+    Ownable,
+    EIP712,
+    ScrollBadgeSelfAttest,
+    ScrollBadgeCustomPayload,
+    ScrollBadgeNoExpiry,
+    ScrollBadgeNonRevocable,
+    ScrollBadgeSingleton
+{
+    /// @dev Thrown when the signature is invalid.
+    error ErrorInvalidSignature();
+
+    /// @notice Emitted when the offchain signer is updated.
+    /// @param oldSigner The address of previous offchain signer.
+    /// @param newSigner The address of current offchain signer.
+    event UpdateOffchainSigner(address indexed oldSigner, address indexed newSigner);
+
+    /// @dev The type hash for issue badge.
+    // solhint-disable-next-line var-name-mixedcase
+    bytes32 private constant _ISSUE_BADGE_TYPEHASH = keccak256("IssueBadge(address recipient,uint256 year)");
```

Comment:
Another way would be to use `AttesterProxy` and check the signature there, any reason you prefer to check it here?

Reply:
didn't notice this contract before. Yeah, we can switch to this.